### ISSUE TITLE
how to compile on M1 mac?

### ISSUE THREAD

**Issue by lucasjinreal** [libxsmm/libxsmm#730]
Got some errors:

```
--------------------------------------------------------------------------------
Compiler: clang 14.0.0, clang 14.0.0, and gfortran 12.2.0
C / C++ target: <compiler default>
Fortran target: <compiler default>
--------------------------------------------------------------------------------
Building a shared library requires to link against BLAS
since a deferred choice is not implemented for this OS.
--------------------------------------------------------------------------------
```

**Comment by jinfagang** [comment]
@hfp thank u for reply.

I just saw there is a thread talk about performance on aarch64, so I thought it support ARM, then I try clone it and then `make`.

I can get the .a files, but cna not build test for got any examples to run.

How can I test libxsmm performance on ARM M1? Or does there any existed data for this?

**Comment by hfp** [comment]
> I just saw there is a thread talk about performance on aarch64, so I thought it supports ARM

Yes, our main revision (still unreleased or soon to be released) supports AArch64 in several flavors.

> I try clone it and then `make`.

This should just work. If not, can you paste the output?

> I can get the .a files

You should get `.a` and `.dynlib` files. Anyway, this seems to work.

> but can not build test for got any examples to run.
> How can I test libxsmm performance on ARM M1?

So, you do not know what to build and run in order to measure performance? Of course, you first need tell me what performance you want to measure or which functionality you are interested on. For example, you might be interested in small matrix multiplications, deep learning primitives such as small convolutions, other layers, or even whole typologies.

> Or does there any existed data for this?

We have not published results yet.

**Comment by jinfagang** [comment]
@hfp  Hi, i thought if libxsmm can run bother on x86 and arm, then I can replace my gemm from openblas to libxsmm.

So I still wonder on some common small convs, how's the speed compares between libxsmm and openblas.

Specifically, I have a matmul, which is  12x32 multiplies 32x128x128, can u compare the benchmark on openblas and libxsmm?

**Comment by hfp** [comment]
> Specifically, I have a matmul, which is 12x32 multiplies 32x128x128, can u compare the benchmark on openblas and libxsmm?

Let me rephrase, you are primarily interested in `MxNxK=32x128x128`? This would refer to an SMM of `[32,128]x[128,128]->[32,128]`. So, I am wondering where the `12x32` comes from? Is the `12` a typo?

Further, let's call the respective matrices `A`, `B`, and `C` such that `AxB->C` (`C=AxB+Beta*C`). Can you let me know the `Beta`, or more important the regime of your matrices/operands? I mean, do you run many different A and B accumulating into few C? How many SMMs can you run consecutively (in a batch at once), or are all SMMs scattered across your code-base? The best way to answer is to tell if A and/or B are "streamed" from DRAM/memory (or potentially held in cache), and if C is accumulating or streamed as well, i.e., individual/different for each/every SMM? If you have sample code modeling your case, this is helpful too.

**Comment by jinfagang** [comment]
@hfp Hi, thanks for your patience, this is an operation takes coefficient from a model outputs 12x32, where 12 is instance num, and 32 is hidden dimension, then it will mupltilies a featuremap which is 32x128x128, the 128x128 is the HW of featuremap, 32 is same hidden dimension.

In python, it like this: `masks_in @ protos.float().view(c, -1)`

So, it should be reshape to -1 for the later matrix, you can compare like: 12x32 multiplies 32x(128*128)

there is not beta, just multiplies, so beta is 0

**Comment by alheinecke** [comment]
closing this issue, if it's still unresolved, please re-open with an updated description.
